keygen/src/keygen.rs

```diff
@@ -1,4 +1,4 @@
-#![allow(clippy::arithmetic_side_effects)]
+#![allow(clippy::arithmetic_side_effects, deprecated)]
```

Comment:
```suggestion
#![allow(clippy::arithmetic_side_effects)]
#![allow(deprecated)]
```

avoid mixing unrelated allows for ease of future removal